Sources/TuistDependencies/Carthage/Utils/CarthageGraphGenerator.swift

```diff
@@ -26,25 +28,37 @@ public final class CarthageGraphGenerator: CarthageGraphGenerating {
             .map { try jsonDecoder.decode(CarthageVersionFile.self, from: $0) }
             .flatMap { $0.allProducts }
 
-        let externalDependencies: [String: [TargetDependency]] = Dictionary(grouping: products, by: \.name)
+        let externalDependencies: [String: [ProjectDescription.TargetDependency]] = Dictionary(grouping: products, by: \.name)
             .compactMapValues { products in
                 guard let product = products.first else { return nil }
 
-                guard let xcFrameworkName = product.container else {
-                    logger.warning("\(product.name) was not added to the DependenciesGraph", metadata: .subsection)
-                    return nil
-                }
-
                 var pathString = ""
                 pathString += Constants.tuistDirectoryName
                 pathString += "/"
                 pathString += Constants.DependenciesDirectory.name
                 pathString += "/"
                 pathString += Constants.DependenciesDirectory.carthageDirectoryName
                 pathString += "/"
-                pathString += xcFrameworkName
 
-                return [.xcframework(path: Path(pathString))]
+                if let xcFrameworkName = product.container {
+                    pathString += xcFrameworkName
+                    return [.xcframework(path: Path(pathString))]
+                }
+
+                if let platforms = platforms {
+                    let paths = platforms.map { $0.carthageDirectory }
+                    let frameworks = paths.map { path -> ProjectDescription.TargetDependency in
+                        pathString += path
+                        pathString += "/"
+                        pathString += product.name
+                        pathString += ".framework"
+                        return .framework(path: Path(pathString))
+                    }
+                    return frameworks
+                }
```

Comment:
If more than one platform is specified in the Dependencies.swift, how is the correct platform selected at generation time?

Reply:
I'm not yet familiar enough with the codebase to know the answer to that. Could you point me to the right direction on where the determination is happening in the code, or where I might add that?

Reply:
The mapping is done in the `TargetDependency+ManifestMapper` file.

The problem is that with the current `DependenciesGraph` structure we don't have enough information to decide which framework should be picked.

It was briefly discussed [here](https://tuistapp.slack.com/archives/C01BYC3ND6U/p1629918751171900) by @laxmorek .

I would suggest discussing the desired solution in an RFC before going in solution mode with a PR

Reply:
Thanks @danyf90, I've seen that discussion and will likely implement one of the options discussed there. @laxmorek and I offered to guide me through working on this change, so I opened up this draft PR for he and I to discuss the work together.